src/coreclr/vm/comdelegate.cpp

```diff
@@ -2166,7 +2159,11 @@ FCIMPL1(PCODE, COMDelegate::GetMulticastInvoke, Object* refThisIn)
 
         ILCodeStream *pCode = sl.NewCodeStream(ILStubLinker::kDispatch);
 
+#ifndef TARGET_X86
+        // x86 has much less available registers comparing to other architectures.
+        // To reduce register pressure and stack spilling, avoid saving invocationCount in local variable.
         DWORD dwInvocationCountNum = pCode->NewLocal(ELEMENT_TYPE_I4);
+#endif // TARGET_X86
```

Comment:
```suggestion
```
I think it is fine to do this for all platforms to keep it simple. Non-x86 platforms have to save an extra register in prolog/epilog for the local variable, so it boils down to whether to micro-optimize for small number of invocations or large number of invocations.

Reply:
The difference looks negligible on x64. We should prefer simplification in this case.